src/main/java/com/icthh/xm/ms/otp/web/rest/OneTimePasswordResource.java

```diff
@@ -75,4 +103,56 @@ public OneTimePasswordResource(OneTimePasswordService oneTimePasswordService) {
         oneTimePasswordService.check(oneTimePasswordCheckDto);
         return ResponseEntity.ok(oneTimePasswordCheckDto);
     }
+
+    /**
+     * POST  /one-time-password/validate : Validate an existing oneTimePassword and redirect page.
+     * @see OneTimePasswordResource#checkOneTimePassword(OneTimePasswordCheckDto)
+     */
+    @PreAuthorize("hasPermission({'oneTimePasswordCheckDto': #oneTimePasswordCheckDto} ,'OTP.ONETIMEPASSWORD.CHECK')")
+    @PostMapping("/one-time-password/validate")
+    @Timed
+    public RedirectView checkOneTimePasswordAndRedirectWithCode(@Valid @RequestBody OneTimePasswordCheckDto oneTimePasswordCheckDto,
+                                                                @RequestHeader(value = "redirect-uri") String redirectUri) throws URISyntaxException {
+        redirectUri = decodeUrl(redirectUri);
+        checkOneTimePassword(oneTimePasswordCheckDto);
+        OneTimePasswordDto dto = oneTimePasswordService.findOne(oneTimePasswordCheckDto.getId()).get();
+        String code =  uaaRepository.getOAuth2Token(ImmutableMap.of(MSISDN, dto.getReceiver()));
+        return new RedirectView(new StringBuilder(redirectUri).append("?code=").append(code).toString(), true);
```

Comment:
mmm, how about method setAttributes ?

Reply:
and what about `setAttributes()` ? Yes we can set http status code via setAttributes and use e.g ModelAndView for redirection ? Is it better ?